### DISCUSSION TITLE
Possible in a playlist to have some songs play automatically one after another, but others to require input to start?

### DISCUSSION THREAD

**Discussion by SirGeoffreyB** [audacious-media-player/audacious#1456] [Q&A] [answered]
I would like to have playlists where I must start a song manually, with a mouse click or hot key, but sometimes have 2 songs play automatically one after another. Is it possible to set that up in a playlist?

Perhaps put two songs as an album and have  the other songs listed individually?
Then, in the playlist have it play a few individual songs manually, then an album with a pair of songs which play one after another without human intervention?

If so, how would I do that?
----------
background:

I'm playing hymns for Church with a very small congregation. Too small for a choir, but they like choir music.
I'm using Audacity to cut out silence, and adjust volume levels so that they are are similar.

I need to play the hymns at appropriate times during the service, so I can't just set up a playlist and have it play them all automatically one after another. But there are times where it would be easiest to have a couple of songs follow one another with a few second pause in between.

Some of those combinations are a given, so I add a few seconds of silence in Audacity & then just join the second song. But there are other places whether the second song will vary from week to week or from service to service and I would rather not use up the disk space to have a bunch of combinations. The ideal would be if I could have the hymns in a playlist, play them one by one at the appropriate times, but occasionally let the playlist play a couple of the hymns back to back automatically, that would be awesome for us.

But I don't know if that is possible... or if it is, how I would do so.

Any help would be most appreciated.

Thank you.

**Comment by mschwendt** [comment]
Not sure why you are worried about using up disk space, but perhaps that is because of preparing customized files with Audacity.

Instead, you could work only with individual songs and multiple playlists. Individual songs that can be part of multiple playlists, and also a "song" of a few seconds of silence, which you can insert where you want to create a pause of X seconds within a playlist. Song entries you copy & paste within a playlist don't take up disk space other than the tiny bits of playlist data. So, for example, to play a song twice or more, copying it into a playlist more than once is the most convenient solution. You would prepare multiple playlists, name them to your liking, and switch between them whenever needed. In case you like to work with simple scripts instead of the graphical user interface, you could set up something with "audtool" as a matter of convenience. Such as a script to choose a named playlist, toggle playlist advance/repeat, possibly select a named song within the playlist, start playback, and so on.

**Reply by SirGeoffreyB** [reply]
Thank you for the reply, and sorry for my late response. Didn't see this 'til now.

The idea for having a 'song' with a few seconds of silence is perfect. Wish I had thought of that.

The reason I was concerned about disk space is that until now I had been using Audacity as a player, and saving the hymns as a large batch of hymns in uncompressed Audacity format. That results in saving a bunch of the hymns multiple times in what I was using as makeshift "playlists". I started resaving the edited hymns individually (& in some cases pairs of hymns) in compressed format and then tried playing them back using VLC. But VLC was playing the hymns one right after another & I didn't want to try and catch them in time to stop hymns from starting at inappropriate times. I couldn't figure out how to play them from a playlist one at a time.

Then I tried Audacious and it worked "out of the box" such that songs would not play until I clicked on them to start. Yay. But there are times when I wanted to play some hymns back to back without having to intervene. Thanks too for the idea on the simple scripts, but actually, I'm not a programmer & don't know how to write scripts.

One option would be to add the silence to the end of one hymn & append the 2nd song on via Audacity, but I would end up with quite a few different combinations and permutations with too many copies of the same hymn. I was hoping there was a way to just have individual copies of each hymn & drop them onto the playlist in the correct order, with individual chunks of silence added, per your idea, and have most of them start when I click with the mouse, but other times to have a couple play automagically one right after the other, then back to one at a time. Not horrible if that can't happen, I was just checking to see if it was possible.

If not, it sounds like you may be recommending having a couple of playlists for a single service? One that would play the songs one at a time when prompted and another that would play a couple of songs one after the other? Then switch back and forth from playlist to playlist to get the same effect?

If so, how would I set one playlist to play songs only when prompted, and another to play one right after another? I couldn't seem to find that in the settings.

Anyway, thanks again for your time and your help, and if you are one of the developers, for writing and sharing this program.

**Comment by mschwendt** [comment]
No, I'm not one of the developers, but just a happy consumer, packager and tester (and external plug-in author, FWIW).

The various playback settings like "repeat", "shuffle" and "stop after this song" are not saved into playlist files, since playlist file formats typically only list a sequence of files (possibly including some metadata about those files to make it available immediately without the need to read/parse all files). Hence those playback settings are global in Audacious and apply to all playlists.

Your usage scenario, as I see it, is similar to that of a DJ. Except that if you don't choose and crossfade between two tracks on-the-fly and spontaneously, your actions are based on prior homework. Like preparing multiple playlists and a cheat sheet in advance. That can get quite complex in my opinion, also if it's done with some audio sequencer or audio editor like Audacity and lots of input via a graphical user interface. The separate "audtool" command-line program's commands can help tremendously with automizing some things within one or multiple running instances of Audacious.

**Reply by SirGeoffreyB** [reply, marked as answer]
> No, I'm not one of the developers, but just a happy consumer, packager and tester (and external plug-in author, FWIW).

Well thank you very much for being so generous with your time and expertise. It is much appreciated, and I am sure that it helps many people. How does one add plug-ins for Audacious? I see [the Audacious plug-ins page](https://github.com/audacious-media-player/audacious-plugins/releases), but I am unfamiliar wit tar.bz2 files & assume that is for Linux (I'm on Windows 11)?
A packager, tester (and external plug-in author) would mean that you are a developer, even if not one of _the_ Audacious developers, no?

> The various playback settings like "repeat", "shuffle" and "stop after this song" are not saved into playlist files...

Thanks. I had not seen the "stop after this song" in the playlist, perhaps because that seems to be the default in Audacious, at least for me, and is the reason Audacious works for me rather than VLC which automatically plays one song right after the other on a playlist. 

After playing around with it a little bit, it seems to me that the various playback settings apparently work on all of songs within a playlist when it is open, but cannot be applied to individual songs within the playlist, correct?

An exception seems to be "stop after this song" which only works on the first song you play after you click that option. I tried to mark the 3rd song down in the list with "stop after this song", then played the first song on the list & it stopped after that one, but when I played the 2nd it automatically played the 3rd song. Having the "no playlist advance" option checked stops each song when it is finished.

I will probably try the workaround of having the playlist with the 'no playlist advance' unchecked so that it will play continuously but use your idea to insert silence between them. On the ones I don't want to play right away I will add clips of 10 seconds or more between them & the next song, which will give the "DJ" (good analogy, btw, that pretty well fits how we will be using this) time to stop the player & cue up the next one. For those where one will follow right after the other I will insert a 3 seconds of silence & let them play as a unit. After that one I will do the longer section of silence to make sure the "DJ" has time to stop it again.

>  Like preparing multiple playlists and a cheat sheet in advance. That can get quite complex

I will need to do multiple playlists in advance. That will be somewhat time consuming, but I have been doing the Audacity files for the full sequence of songs for several months. I have just been doing 1 week at a time, so just one playlist per week, plus a few here & there for special occasions. Saving the individual hymns in AAC format again after trimming the silence at the beginning & leveling the sound levels will be an extra step, but it will save disk space & will be easier for others to use for playback.

> The separate "audtool" command-line program's commands can help tremendously with automizing some things within one or multiple running instances of Audacious.

That sounds interesting. I searched & found [this page on it](https://manpages.ubuntu.com/manpages/focal/man1/audtool.1.html). It looks like it is an Ubuntu program, however. I had a programming class several decades ago, but it would probably be a bit of a learning curve for me to make it useful for me anyway. I do appreciate the tip, though.

Do you have a page for your plug-ins?

Thanks again for your time and your help.

**Comment by mschwendt** [comment]
On Windows, the prebuilt Audacious come as an .EXE installer which installs a high number of plug-ins, because many of them are essential. Inside the graphical menu, the plug-ins are found under "File > Settings > Plug-ins".

Audtool is available on Windows, too, as audtool.exe alongside audacious.exe. By default, it's found in a path like: C:\Program Files (x86)\Audacious\bin\

You would run it from Windows Command Prompt (or any similar program). Similarly, you could start multiple instances of Audacious via "audacious.exe -1", "audacious.exe -2", and so on. Your playback settings within those running instances may differ then. Pretty convenient.

**Reply by SirGeoffreyB** [reply]
Thank you very much.

I marked one of my own posts as an answer, but I was just recapping answers you had given to me. I could mark all of your posts as answers, but it won't let me. ;-).

Thanks again for all of your time & help.
